clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleHttpLoadBalancerCachingAgent.groovy

```diff
@@ -0,0 +1,381 @@
+        loadBalancers: loadBalancers,
+        targetProxyRequest: targetProxyRequest,
+        urlMapRequest: urlMapRequest,
+        backendServiceRequest: backendServiceRequest,
+        httpHealthChecksRequest: httpHealthChecksRequest,
+        groupHealthRequest: groupHealthRequest,
+    )
+    compute.globalForwardingRules().list(project).queue(forwardingRulesRequest, callback)
+
+    executeIfRequestsAreQueued(forwardingRulesRequest)
+    executeIfRequestsAreQueued(targetProxyRequest)
+    executeIfRequestsAreQueued(urlMapRequest)
+    executeIfRequestsAreQueued(backendServiceRequest)
+    executeIfRequestsAreQueued(httpHealthChecksRequest)
+    executeIfRequestsAreQueued(groupHealthRequest)
+
+    return loadBalancers
+  }
+
+  CacheResult buildCacheResult(ProviderCache _, List<GoogleLoadBalancer> googleLoadBalancers) {
+    log.info "Describing items in ${agentType}"
+
+    def cacheResultBuilder = new CacheResultBuilder()
+
+    googleLoadBalancers.each { GoogleLoadBalancer loadBalancer ->
+      def loadBalancerKey = Keys.getLoadBalancerKey(loadBalancer.region,
+          loadBalancer.account,
+          loadBalancer.name)
+      def instanceKeys = loadBalancer.healths.collect { GoogleLoadBalancerHealth health ->
+        // Http load balancers' region is "global", so we have to determine the instance region from its zone.
+        def instanceZone = health.instanceZone
+        def instanceRegion = instanceZone.substring(0, instanceZone.lastIndexOf("-"))
+        Keys.getInstanceKey(accountName, instanceRegion, health.instanceName)
+      }
+
+      cacheResultBuilder.namespace(LOAD_BALANCERS.ns).keep(loadBalancerKey).with {
+        attributes = objectMapper.convertValue(loadBalancer, ATTRIBUTES)
+      }
+      instanceKeys.each { String instanceKey ->
+        cacheResultBuilder.namespace(INSTANCES.ns).keep(instanceKey).with {
+          relationships[LOAD_BALANCERS.ns].add(loadBalancerKey)
+        }
+      }
+    }
+
+    log.info "Caching ${cacheResultBuilder.namespace(LOAD_BALANCERS.ns).keepSize()} load balancers in ${agentType}"
+    log.info "Caching ${cacheResultBuilder.namespace(INSTANCES.ns).keepSize()} instance relationships in ${agentType}"
+
+    cacheResultBuilder.build()
+  }
+
+  @Override
+  boolean handles(OnDemandAgent.OnDemandType type, String cloudProvider) {
+    type == OnDemandAgent.OnDemandType.LoadBalancer && cloudProvider == GoogleCloudProvider.GCE
+  }
+
+  @Override
+  Collection<Map> pendingOnDemandRequests(ProviderCache providerCache) {
+    return []
+  }
+
+  /**
+   * This is a "simpleton" way of handling on-demand cache requests. Load Balancer mutation (and thus the need for
+   * cache refreshing) is not as common or complex as server group cache refreshes.
+   *
+   * This implementation has the potential for race condition between handle() and loadData(), which may
+   * cause "flapping" in the UI. lwander@ has plans to make an abstract solution for this race condition, so this impl
```

Comment:
I don't think so, I saw the LB details still flapping in the UI on cache refresh cycles.